app/src/main/java/com/vero/tasky/agenda/presentation/components/ReminderComponent.kt

```diff
@@ -21,46 +19,75 @@ import com.vero.tasky.agenda.presentation.model.ReminderRange
-                .clickable(enabled = isEditable) { onReminderClick() }) {
+    var isReminderDropDownAreVisible by remember {
+        mutableStateOf(false)
+    }
+
+    Column {
+        Box(modifier = modifier.fillMaxWidth().clickable(isEditable) { isReminderDropDownAreVisible = true }) {
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.SpaceBetween,
+            ) {
+                Box(modifier = Modifier
+                    .clip(RoundedCornerShape(5.dp))
+                    .width(35.dp)
+                    .height(35.dp)
+                    .background(reminderBackgroundColor)
+                    .clickable(enabled = isEditable) { isReminderDropDownAreVisible = true }) {
+                    Icon(
+                        imageVector = Icons.Outlined.Notifications,
+                        contentDescription = stringResource(id = R.string.choose_reminder),
+                        tint = MaterialTheme.colors.onTextFieldIcon,
+                        modifier = Modifier.align(Alignment.Center)
+                    )
+                }
+                Spacer(modifier = Modifier.width(16.dp))
+                Text(
+                    text = stringResource(id = reminderRange.label),
+                    style = MaterialTheme.typography.Inter400Size16,
+                    color = MaterialTheme.colors.primary
+                )
+            }
+            if (isEditable) {
                 Icon(
-                    imageVector = Icons.Outlined.Notifications,
-                    contentDescription = stringResource(id = R.string.add_visitor),
-                    tint = MaterialTheme.colors.onTextFieldIcon,
-                    modifier = Modifier.align(Alignment.Center)
+                    imageVector = Icons.Default.ArrowRight,
+                    contentDescription = stringResource(id = R.string.edit),
+                    tint = MaterialTheme.colors.primary,
+                    modifier = Modifier.align(Alignment.CenterEnd)
                 )
             }
-            Spacer(modifier = Modifier.width(16.dp))
-            Text(
-                text = stringResource(id = reminderRange.label),
-                style = MaterialTheme.typography.Inter400Size16,
-                color = MaterialTheme.colors.primary
-            )
         }
-        if (isEditable) {
-            Icon(
-                imageVector = Icons.Default.ArrowRight,
-                contentDescription = stringResource(id = R.string.edit),
-                tint = MaterialTheme.colors.primary,
-                modifier = Modifier.align(Alignment.CenterEnd)
-            )
+        if (isReminderDropDownAreVisible) {
+            DropdownMenu(
+                expanded = true,
+                onDismissRequest = {
+                    isReminderDropDownAreVisible = false
+                },
+            ) {
+                ReminderRange.values().forEach { reminderRange ->
```

Comment:
This is something really minor

But on each recomposition, you'll be redrawing this for loop. Since "ReminderRange.values()" will recalculate

You could have the "ReminderRange.values()" on a remember state, so it only get's calculated once

(Maybe it needs to be derivedStateOf)